docs/proposals/scheduling/20230314-gpu-topology-aware-scheduling.md

```diff
@@ -0,0 +1,275 @@
+
+The following is a matrix describing the bandwidth between 8 GPU cards on a node, and the unit of value is GB/s:
+```
+Bandwidth Matrix:
+       gpu_0   gpu_1   gpu_2   gpu_3   gpu_4   gpu_5   gpu_6   gpu_7
+gpu_0  750.48  48.39   48.33   96.41   15.77   15.52   96.40   15.74
+gpu_1  48.39   753.38  96.46   48.38   4.64    16.93   16.98   96.39
+gpu_2  48.38   96.25   751.92  96.48   48.39   17.57   17.59   16.72
+gpu_3  96.25   48.39   96.43   750.48  15.45   48.39   15.88   14.62
+gpu_4  5.00    16.81   48.38   15.98   755.56  96.39   48.38   96.44
+gpu_5  15.80   16.93   17.50   48.39   96.25   751.92  96.23   48.38
+gpu_6  96.42   16.75   17.47   15.89   48.35   96.28   754.10  48.33
+gpu_7  15.65   96.20   16.77   15.71   96.25   48.38   48.33   754.83
+```
+If a distributed training job has 2 Pods, and each Pod requests 2 GPU cards, then the [gpu0, gpu1, gpu2, gpu3] combination should be selected first rather than the [gpu0, gpu1, gpu2, gpu5] combination, because the former The bottleneck bandwidth is 48.33 (bottleneck bandwidth refers to the minimum bandwidth of any two GPU connections in a group of GPU cards), while the bottleneck bandwidth of the latter is 4.64. If the latter is allocated to the training job, it will greatly affect the training time.
+### Goals
+
+1. A scheduling plugin is provided, which considers the bandwidth between GPU cards when allocating GPU cards for pods and preferentially selects the GPU cards combination with large bottleneck bandwidth.
+2. The scheduling plugin supports allocating GPU cards to individual Pods and groups of Pods.
+3. Topology-aware scheduling tries to select the optimal combination of GPU cards currently available on the node for the training job rather than a mandatory behavior; that is to say, the GPU group allocated for the training job may also be the worst combination.
+4. If a node cannot place all the pods of the training job, it will try to place these pods with the fewest nodes to avoid node resource fragmentation.
+### Non-Goals/Future Work
+
+1. In this proposal, it is assumed that a training job can tolerate some pods running on the node first while the remaining pods are pending. If the training job cannot tolerate this situation, the GPU topology plugin needs to be used in conjunction with the gang plugin to implement All Or Nothing scheduling; that is, this solution does not implement the All Or Nothing scheduling logic.
```

Comment:
I have question for this part, why we need to say we let some pods running on the node first?  may be we should declare that we may need podgroup to describe a group of pod, and find a best schedule-result of them. whether the pods need launch together or not after scheduling is not very related to the topic, this is my opinion.